pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -94,6 +94,11 @@ stdenv.mkDerivation ({
 
   patches = [
     ./env_var_for_system_dir.patch
+    (fetchpatch {
```

Comment:
Should this only be in `patches` if `waylandSupport`?

Reply:
We should have less conditional patches as that makes it harder to have silent breakage. Usually those features should only be disabled via the build system and not via conditional patches.

Reply:
After reading the patch, which doesn't support making pipewire optional, I am actually in favor of just dropping it or making it at least non-default. I am even less motivated to maintain this now then I was before looking at the patch. It seems like it needs to at least contain one more feature flag check in `media/webrtc/trunk/webrtc/modules/desktop_capture/desktop_capture_generic_gn/moz.build`. Probably the same that was applied in `media/webrtc/trunk/webrtc/modules/desktop_capture/BUILD.gn`. And `pipewire` and `spa` should be discovered via pkg_config (like in the other case) not just via hardcoded paths.